Sources/Router/Colorization/OARouteColorize.mm

```diff
@@ -245,40 +245,126 @@ - (void) correctElevations:(NSMutableArray<NSNumber *> *)elevations
     }
 }
 
-- (QList<OsmAnd::FColorARGB>) getResult
+- (NSArray<OARouteColorizationPoint *> *)getResult
 {
-    QList<OsmAnd::FColorARGB> result;
-    for (int i = 0; i < _values.count; i++)
+    NSMutableArray<OARouteColorizationPoint *> *result = [NSMutableArray array];
+    for (int i = 0; i < _latitudes.count; i++)
     {
-        result.push_back([self getColorByValue:_values[i].doubleValue]);
+        [result addObject:[[OARouteColorizationPoint alloc] initWithIdentifier:i lat:_latitudes[i].doubleValue lon:_longitudes[i].doubleValue val:_values[i].doubleValue]];
     }
+    [self setColorsToPoints:result];
     return result;
 }
 
-- (OsmAnd::FColorARGB) getColorByValue:(double)value
+- (QList<OsmAnd::FColorARGB>)getResultQList
+{
+    NSArray<OARouteColorizationPoint *> *result = [self getResult];
+    QList<OsmAnd::FColorARGB> qResult;
+    for (OARouteColorizationPoint *colorizationPoint in result)
+    {
+        qResult.append(OsmAnd::ColorARGB(colorizationPoint.color));
+    }
+    return qResult;
+}
+
+- (NSArray<OARouteColorizationPoint *> *)getSimplifiedResult:(NSInteger)simplificationZoom
```

Comment:
It looks like `getSimplifiedResult`, `simplify` and perhaps other new methods are redundant. Why do we need them?